hudi-common/src/main/java/org/apache/hudi/common/model/HoodieTableQueryType.java

```diff
@@ -30,7 +30,7 @@
  * </ol>
  */
 public enum HoodieTableQueryType {
-  QUERY_TYPE_SNAPSHOT,
-  QUERY_TYPE_INCREMENTAL,
-  QUERY_TYPE_READ_OPTIMIZED
+  SNAPSHOT,
```

Comment:
this might be break things for users. I mean, this is what existing users might be using in their pipelines to read hudi tables. Let's be cautious in these renames.

Reply:
This has been introduced in one of the PRs in this stack

Reply:
I am not sure if we can do this. Users might have to fix all of their existing pipelines w/ these new variables. atleast for read optimized query and incremental query pipelines.

Reply:
I see deduceQueryType in SparkHoodieTableFileIndex. so if we are converting the old vars to new ones, we should be good. Can you do some sanity testing w/ quick start guide that old variables work w/ this patch too.

Reply:
and does the variable conversion happen in all code paths. for eg, there is a config where in you can disable fileIndex altogether. So, presto, trino, hive connector, beeline, spark datasource from all possible paths, ensure old to new is taken care of.

Reply:
for eg, do we need to add it [here](https://github.com/apache/hudi/blob/973087f385f0e42bd42e2d57008a756e182041ff/hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/DataSourceOptions.scala#L851)

Reply:
@yihua : may I know why this is marked as resovled ?

Reply:
Not sure i understand where the disconnect is: there's no old variables -- this enum has been introduced in one of my preceding PRs in this same stack. As such, nobody could possibly be using it unless they depend on snapshot artifacts, which i deem unlikely.

Moreover, internal components that are not annotated as _public API_ are implicitly considered private components and do not the same compatibility guarantees attached as do the Public APIs, and therefore should not be dependent upon as these could change w/o notice.

Reply:
ok, my bad totally. I got confused with DataSourceReadOptions.QUERY_TYPE_SNAPSHOT_OPT_VAL and other variables. we are good here.

Reply:
@nsivabalan looks like we are good here.  These are totally new enums, not there in 0.10.0 release.

Reply:
I don't blame @nsivabalan for the PTSD :) Good to see the paranoia. Thats the first thought that flashed into my mind as well